app/src/main/java/org/ea/sqrl/activites/UrlLoginActivity.java

```diff
@@ -132,36 +130,19 @@ public void afterTextChanged(Editable s) {
             long currentId = SqrlApplication.getCurrentId(this.getApplication());
 
             if(currentId != 0) {
-                doLogin(storage, txtLoginPassword, false, true, null,this);
+                doLogin(storage, txtLoginPassword, false, useCps, true, null,this);
             }
         });
 
-        if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.P && storage.hasBiometric()) {
+        if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.P && storage.hasBiometric() && !ACTION_LOGON.equals(intent.getAction())) {
```

Comment:
I think we should move the whole biometric stuff into the first `if (!ACTION_LOGON.equals(intent.getAction())) { 
}` block above. This way, we don't need a second test for the action, and we also eliminate the variable scoping issues. So you could leave stuff like `final String domain = sqrlMatcher.group(1);` in and actually use it in the biometric block, instead of having to fall back on the nondescript `mSqrlMatcher.group(1)`. What do you think?

Reply:
Rather than struggle with this, I just "made it work", which, of course, isn't the right way to proceed.  If I had seen the opportunity you point out of moving all the biometric stuff in to the first block, I would have done it, but the idea didn't come to me. 

I set this PR to "allow collaborators to change" or something like that.  Would you / could you make this change @alexhauser?

Reply:
I guess I'll have to clone your forked repo and then I should be able to push to your feature branch. I'll have a look asap.

Reply:
Thinking about this again, I guess I'll leave this as-is for now and post a follow-up PR with the mentioned optimization later instead of squeezing it in here. Small diffs ftw.